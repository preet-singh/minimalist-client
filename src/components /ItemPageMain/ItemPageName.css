
.ItemPageMain {
    background-color: #c2c2d6;
    height: 100%;
    padding: 10px;
    width: 100%;
  }
  
  .ItemPageMain_description {
    color: #3a3939;
    max-width: 380px;
  }
  
  .addNoteOrFolder {
    color: white;
    text-align: center;
  }
  
  .validationElement {
    color: red;
  }