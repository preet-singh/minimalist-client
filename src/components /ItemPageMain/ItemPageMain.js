import React from 'react';
import Item from '../Item/Item';
import InventoryContext from '../../InventoryContext';
import { findItem } from '../item-helpers';
import PropTypes from 'prop-types';
import './ItemPageName.css';

export default class ItemPageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  };
  static contextType = InventoryContext;

  handleDeleteItem = itemId => {
    this.props.history.push(`/`)
  };

  render() {
    const { items=[] } = this.context
    const { itemId } = this.props.match.params
    const item = findItem(items, Number(itemId)) || { item_description: 'test desc', item_action: 'test action'}
    console.log(item);
    return (
      <section className='ItemPageMain'>
        <Item
          id={item.id}
          name={item.item_name}
          onDeleteItem={this.handleDeleteItem}
        />
        <div className='ItemPageMain_description'>
          {item.item_description.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>Description: {para}</p>
          )}
        </div>
        <div className='ItemPageMain_action'>
          {item.item_action.split(/\n \r|\n/).map((para, i) =>
            <p key={i}>Action: {para}</p>
          )}
        </div>
      </section>
    )
  }
}

ItemPageMain.propTypes = {
  match: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
}
