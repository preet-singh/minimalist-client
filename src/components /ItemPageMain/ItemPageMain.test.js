import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemPageMain from './ItemPageMain'

describe(`ItemPageMain component`, () => {
  it('renders a .ItemPageMain by default', () => {
    const wrapper = shallow(<ItemPageMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Item with item prop', () => {
    const props = {
      match: {
        params: {
          itemId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }
    const context = {
      items: [{
        id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
        name: `Dogs`,
        // inventoryId: b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1,
        description: "Corporis accusamus placeat.\n \rUnde."
      }]
    }
    const item = shallow(<ItemPageMain {...props} />, context)
      .find('Item')
    expect(toJson(item)).toMatchSnapshot()
  })

  it(`splits the description by \\n or \\n\\r, with a p foreach`, () => {
    const props = {
      match: {
        params: {
          itemId: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }

    const itemsContextWithDifferentDescription = [
      {
        items: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            description: "description with n r.\n \rafter n r.",
          }
        ]
      },
      {
        items: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            description: "description with n.\nafter."
          }
        ]
      }
    ]

    itemsContextWithDifferentDescription.forEach(context => {
      const description = shallow(<ItemPageMain {...props} />, context)
        .find('ItemPageMain__description')
      expect(toJson(description)).toMatchSnapshot()
    })
  })
})
