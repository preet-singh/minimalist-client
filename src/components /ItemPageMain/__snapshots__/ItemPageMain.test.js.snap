// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPageMain component renders a .ItemPageMain by default 1`] = `
<section
  className="ItemPageMain"
>
  <Item
    onDeleteItem={[Function]}
  />
  <div
    className="ItemPageMain_description"
  >
    <p
      key="0"
    >
      test desc
    </p>
  </div>
  <div
    className="ItemPageMain_action"
  >
    <p
      key="0"
    >
      test action
    </p>
  </div>
</section>
`;

exports[`ItemPageMain component renders a Item with item prop 1`] = `
<Item
  onDeleteItem={[Function]}
/>
`;

exports[`ItemPageMain component splits the description by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`ItemPageMain component splits the description by \\n or \\n\\r, with a p foreach 2`] = `null`;
